compiler-plugin/src/main/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqServiceValidator.java

```diff
@@ -99,45 +98,48 @@ private void validateAttachPoint(SyntaxNodeAnalysisContext context) {
                 }
-            } else {
-                if (attachPoint.get().kind() != ServiceAttachPointKind.STRING_LITERAL) {
-                    if (serviceDeclarationSymbol.annotations().isEmpty()) {
-                        context.reportDiagnostic(PluginUtils.getDiagnostic(
-                                CompilationErrors.INVALID_SERVICE_ATTACH_POINT,
-                                DiagnosticSeverity.ERROR, serviceDeclarationNode.location()));
-                    } else {
-                        validateAnnotation(symbolList.get(0), serviceDeclarationNode.location(), context);
-                    }
+            } else if (!hasServiceConfig(annotations)) {
+                if (serviceNameAttachPoint.isEmpty()) {
+                    // Case 1: No service name and no annotation
+                    reportError(context, CompilationErrors.NO_ANNOTATION, serviceDeclarationNode);
+                } else if (serviceNameAttachPoint.get().kind() != ServiceAttachPointKind.STRING_LITERAL) {
+                    // Case 2: Service name is not a string and no annotation
+                    reportError(context, CompilationErrors.INVALID_SERVICE_ATTACH_POINT, serviceDeclarationNode);
                 }
             }
         }
     }
 
-    private void validateAnnotation(AnnotationSymbol annotationSymbol, Location location,
-                                    SyntaxNodeAnalysisContext context) {
-        Optional<ModuleSymbol> moduleSymbolOptional = annotationSymbol.getModule();
-        if (moduleSymbolOptional.isEmpty()) {
-            context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.INVALID_ANNOTATION,
-                    DiagnosticSeverity.ERROR, location));
-        } else {
-            ModuleSymbol moduleSymbol = moduleSymbolOptional.get();
-            if (!moduleSymbol.id().orgName().equals(PluginConstants.PACKAGE_ORG) ||
-                    !moduleSymbol.id().moduleName().equals(PluginConstants.PACKAGE_PREFIX)) {
-                context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.INVALID_ANNOTATION,
-                        DiagnosticSeverity.ERROR, location));
+    private boolean hasServiceConfig(List<AnnotationSymbol> annotationSymbols) {
+        boolean flag = false;
+        for (AnnotationSymbol annotationSymbol : annotationSymbols) {
+            Optional<ModuleSymbol> moduleSymbolOptional = annotationSymbol.getModule();
+            if (moduleSymbolOptional.isPresent()) {
+                ModuleSymbol moduleSymbol = moduleSymbolOptional.get();
+                if (moduleSymbol.id().orgName().equals(PluginConstants.PACKAGE_ORG) ||
+                        moduleSymbol.id().moduleName().equals(PluginConstants.PACKAGE_PREFIX)) {
+                    // not checking name as rabbitmq has only two annotations and only one is allowed on services.
+                    flag = true;
+                }
             }
         }
+        return flag;
```

Comment:
* No need for a separate `flag` variable. We can directly return `true` if we find the correct annotation. No need to iterate through the whole list, if we found the correct one.
* When using the `equals` check, always better to use the `equals()` method on the constant, will prevent NPEs. 
* (Subjective take) I don't think using `Optional<>` variables is a good thing. Returning `Optional` is one thing, but assigning variables to `Optional` doesn't look nice.
* The inner-most `if` condition should be `&&`, not `||`, right? We should check whether both the org and module name are matching. 

```suggestion
        for (AnnotationSymbol annotationSymbol : annotationSymbols) {
            if (annotationSymbol.getModule().isPresent()) {
                ModuleSymbol moduleSymbol = annotationSymbol.getModule().get();
                if (PluginConstants.PACKAGE_ORG.equals(moduleSymbol.id().orgName()) &&
                        PluginConstants.PACKAGE_PREFIX.equals(moduleSymbol.id().moduleName())) {
                    // not checking name as rabbitmq has only two annotations and only one is allowed on services.
                    return true;
                }
            }
        }
        return false;
```